presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/AddIntermediateAggregations.java

```diff
@@ -202,7 +205,10 @@ private PlanNode addGatheringIntermediate(AggregationNode aggregation, PlanNodeI
         ImmutableMap.Builder<Symbol, Aggregation> builder = ImmutableMap.builder();
         for (Map.Entry<Symbol, Aggregation> entry : assignments.entrySet()) {
             // Should only have one input symbol
-            Symbol input = getOnlyElement(SymbolsExtractor.extractAll(entry.getValue().getCall()));
+            Aggregation aggregation = entry.getValue();
+            checkArgument(aggregation.getArguments().size() == 1 && !aggregation.getOrderBy().isPresent() && !aggregation.getFilter().isPresent(),
```

Comment:
nit: if one parameter per line is used, I personally prefer to have new line for the first parameter as well, e.g. 
```java
checkArgument(
           aggregation.getArguments().size() == 1 && !aggregation.getOrderBy().isPresent() && !aggregation.getFilter().isPresent(),
           " Aggregation should only have one argument and should have no order by  or filter to be able to rewritten to intermediate form");
```